### PR TITLE
chore(doc): add default values & parameter types

### LINKED ISSUE #473: Documentation: missing default values in jsdoc
jsdoc (all options) is missing the default values we have in our documentation.

### PR DESCRIPTION
**Pull request by redox**
- use the jsdoc2md tool to generate the `instantsearch()` and
   `addWidget` documentation as well
 - update the default template to display the default value & type

Fix #473

### CODE REVIEW

File: scripts/doc/gen-widget-doc.js [hunk 1]
```diff
@@ -3,16 +3,31 @@ let dmd = require('dmd');
...
+let ctorToFunction = new Transform({decodeStrings: false});
+ctorToFunction._transform = (chunk, encoding, done) => {
+  let data = eval(chunk.toString()); // eslint-disable-line
+  data.forEach(function(e) {
+    if (e.kind === 'constructor') {
+      e.kind = 'function';
+    }
+  });
+  done(null, JSON.stringify(data));
+};
```
Comment (redox) [inline scripts/doc/gen-widget-doc.js]: I had a hard time figuring out how to use the same template for both `function` and `constructor` and I ended up with that. Probably not perfect but does the job.

Comment (vvo) [inline scripts/doc/gen-widget-doc.js]: WFM as it's not crippling the underlying code (implementation of InstantSearch), maybe we could be able to override the type with a specific tag, dunno

Comment (redox) [inline scripts/doc/gen-widget-doc.js]: Yes, but it was way more complex than I thought so I ended up with that :)

**Comment by vvo** [comment]
> Yes, but it was way more complex than I thought so I ended up with that :)

I removed all of it with @function

giphy

**Comment by redox** [comment]
:+1:

**Pull request merged**